js/index.js

```diff
@@ -1,42 +1,78 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  const price = product.querySelector('.price span').textContent;
+  const quantity = product.querySelector('.quantity input').value;
+  const subtotal = product.querySelector('.subtotal span');
+  
+  subtotal.textContent = price * quantity; // Updates subtotal on the page
+  return Number(subtotal.textContent);
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
-  // ITERATION 2
-  //... your code goes here
+  // ITERATION 2 + 3
+  let sum = 0;
+  const products = document.querySelectorAll('.product');
+  for (let i = 0; i < products.length; i++)
+    sum += updateSubtotal(products[i]); // Updates the product subtotal and increases the total price
 
-  // ITERATION 3
-  //... your code goes here
+  const total = document.querySelector('#total-value span');
+  total.textContent = `${sum}`; // Updates total price
 }
 
 // ITERATION 4
 
 function removeProduct(event) {
-  const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  const target = event.currentTarget.parentNode.parentNode;
+  
+  target.parentNode.removeChild(target); // Removes the item from the table
+  calculateAll();
 }
 
 // ITERATION 5
 
 function createProduct() {
-  //... your code goes here
+  const parent = document.getElementsByTagName('tbody');
+  const product = document.querySelectorAll('.create-product input');
+  const name = product[0].value; // Name of the new product
+  const price = product[1].value; // Price of the new product
+   
+  const newProduct = `
+  <tr class="product">
+          <td class="name">
+            <span>${name}</span>
+          </td>
+          <td class="price">$<span>${price}</span></td>
+          <td class="quantity">
+            <input type="number" value="0" min="0" placeholder="Quantity" />
+          </td>
+          <td class="subtotal">$<span>0</span></td>
+          <td class="action">
+            <button class="btn btn-remove">Remove</button>
+          </td>
+        </tr>
+  `;
+
+  parent[0].innerHTML += newProduct; // Adds a new product to the table
```

Comment:
A small remark here. When you add a new product, the quantity of the existing products goes back to 0. I think because the html has 0 as the quantity of the products and in this line it just replaces what is in the DOM. You could also use `insertRow()`.
```
  let newRow = parent[0].insertRow()
  newRow.className = 'product'
  newRow.innerHTML = newProduct;
```
This will keep the remaining products unchanged.